File: LintCode/158_two-strings-are-anagrams/two-strings-are-anagrams.java
/*
@Copyright:LintCode
@Author:   siyang
@Problem:  http://www.lintcode.com/problem/two-strings-are-anagrams
@Language: Java
@Datetime: 16-10-28 02:34
*/

public class Solution {
    /**
     * @param s: The first string
     * @param b: The second string
     * @return true or false
     */
    public boolean anagram(String s, String t) {
        // write your code here
        if(s == null && t == null)
            return true;
        if(s == null || t == null || s.length() != t.length())
            return false; 
        int [] count = new int [256];
        for(int i = 0 ; i < s.length() ; i ++){
            count[s.charAt(i)] ++;
        }
        
        for(int i = 0 ; i < t.length() ; i ++){
            if( -- count[t.charAt(i)] < 0)
                return false;
        }
        return true;
        
    }
};

File: LintCode/158_two-strings-are-anagrams/two-strings-are-anagrams.py
# coding:utf-8
'''
@Copyright:LintCode
@Author:   siyang
@Problem:  http://www.lintcode.com/problem/two-strings-are-anagrams
@Language: Python
@Datetime: 16-11-05 19:24
'''

class Solution:
    """
    @param s: The first string
    @param b: The second string
    @return true or false
    """
    def anagram(self, s, t):
        # write your code here
        ss = [i for i in s]
        tt = [j for j in t]
        ss.sort()
        tt.sort()
        return "".join(ss) == "".join(tt)